Detectors/EMCAL/base/src/Geometry.cxx

```diff
@@ -1762,6 +1767,25 @@ void Geometry::SetMisalMatrix(const TGeoHMatrix* m, Int_t smod) const
   }
 }
 
+void Geometry::SetMisalMatrixFromCcdb(const char* path, int timestamp) const
+{
+  LOG(info) << "Using CCDB to obtain EMCal alignment.";
+  auto& mCcdbManager = o2::ccdb::BasicCCDBManager::instance();
+  mCcdbManager.setURL(o2::base::NameConf::getCCDBServer());
+  mCcdbManager.setTimestamp(timestamp);
```

Comment:
do you really need to set a timestamp and URL here? Beware, that BasicCCDBManager is a convenience singleton which is supposed to be initialized by the simulation itself. Clients should only use it as read-access. If you set url and timestamp here it will potentially affect everyone else. Note also that "mCcdbManager" gives the impression to be a class member which it is not. Just `const auto& ccdbmgr` would be better.

I agree that this is partially bad design. Could introduce a "locking" mechanism which prevents misuse of BasicCCDBManager.

Reply:
Hey @sawenzel 
I am unsure if it is needed, since I have limited knowledge of the ccdb manager. All I want is to just get the object I uploaded into the CCDB. If this is possible without setting the URL, I would change it. I gan also use `getForTimeStamp` to not set the timestamp. I am just not aware how to ensure that the correct URL is used without setting it yourself. Otherwise it might be better to just not use the singleton, what do you think?

Also please keep in mind that this will not be called during reconstruction, only later on hyperloop.

Reply:
We use the production URL for CCDB. If your objects are there it should be fine. Timestamp should also be defined by the simulation framework. So it should be ok.

If you need full manual control, then using CcdbApi directly (instead of BasicCCDBManager) is more appropriate.

In an O2DPG MC simulation you can actually check which objects are downloaded at which time. There is a `ccdb` folder that contains a log with this information.